crates/fuel-core/src/schema/tx/assemble_tx.rs

```diff
@@ -912,23 +912,25 @@ where
         let mut total_base_asset = 0u64;
 
         for input in self.tx.inputs() {
-            let Some(amount) = input.amount() else {
-                continue;
-            };
-            let Some(asset_id) = input.asset_id(&base_asset_id) else {
-                continue;
-            };
-            let Some(owner) = input.input_owner() else {
-                continue;
-            };
+            if !input.is_message_data_signed() && !input.is_message_data_predicate() {
```

Comment:
Nit: It would be helpful to break this out into a separate function with an informative name. Something like `input.is_spendable()` perhaps.